chrome/background.js

```diff
@@ -95,6 +95,44 @@ function onMessage(request, sender, sendResponse) {
       localStorage.getItem("use_fuzzy_search") != "false";
     sendResponse({ use_fuzzy_search: use_fuzzy_search });
   }
+
+  // spawn a new tab with pre-provided credentials
+  if (request.action == "launch") {
+    chrome.tabs.create({url: request.url}, function (tab) {
+      var authAttempted = false;
+      chrome.webRequest.onAuthRequired.addListener(
+        function authListener(requestDetails) {
+          // only supply credentials if this is the first time for this tab
+          if (authAttempted) {
+            return {};
+          }
+          authAttempted = true;
+          // remove event listeners once tab loading is complete
+          chrome.tabs.onUpdated.addListener(function statusListener(tabId, info) {
+              if (info.status === "complete") {
```

Comment:
@erayd do you think this code should also compare `tabId` with the `tab.id` before removing the listeners?

Reply:
It doesn't need to, because the listeners are specific to the tab anyway.

There is a security problem, but that's not it. I'm filing an issue now.

Reply:
Are you sure that this doesn't register a listener that fires for _all tabs_? Maybe it's just a weird API, but the tab object is not used when registering the listener, and the `tabId` is provided as a first argument for a reason, both of these lead me to think that this will fire for all tabs...

I know that this is not about the security problem, it's just what I accidentally stumbled upon 🙂

Reply:
Ooh, good point. The `onAuthRequested` handler is specific to the tab, but the status listener is not. My bad.

I'm going to roll the security fix PR right now - I feel quite bad I didn't notice that in the first place; I'm normally more call than that. I will fix this at the same time.

Reply:
Thank you!